UefiTestingPkg/FunctionalSystemTests/MemoryProtectionTest/App/MemoryProtectionTestApp.c

```diff
@@ -996,51 +1038,88 @@ UefiPoolGuard (
-    MemoryProtectionContext.TestProgress++;
-    SaveFrameworkState (&MemoryProtectionContext, sizeof (MEMORY_PROTECTION_TEST_CONTEXT));
-    SetBootNextDevice ();
+      //
+      // Memory type rHardwareNxProtections to the bitmask for the PcdHeapGuardPoolType,
+      // we need to RShift 1 to get it to reflect the correct EFI_MEMORY_TYPE.
+      //
+      Status = gBS->AllocatePool ((EFI_MEMORY_TYPE)MemoryProtectionContext.TargetMemoryType, AllocationSize, (VOID **)&ptr);
 
-    //
-    // Memory type rHardwareNxProtections to the bitmask for the PcdHeapGuardPoolType,
-    // we need to RShift 1 to get it to reflect the correct EFI_MEMORY_TYPE.
-    //
-    Status = gBS->AllocatePool ((EFI_MEMORY_TYPE)MemoryProtectionContext.TargetMemoryType, AllocationSize, (VOID **)&ptr);
+      if (EFI_ERROR (Status)) {
+        UT_LOG_WARNING ("Memory allocation failed for type %a of size %x - %r\n", MEMORY_TYPES[MemoryProtectionContext.TargetMemoryType], AllocationSize, Status);
+        return UNIT_TEST_SKIPPED;
+      }
 
-    if (EFI_ERROR (Status)) {
-      UT_LOG_WARNING ("Memory allocation failed for type %a of size %x - %r\n", MEMORY_TYPES[MemoryProtectionContext.TargetMemoryType], AllocationSize, Status);
-      return UNIT_TEST_SKIPPED;
-    } else {
       PoolTest ((UINT64 *)ptr, AllocationSize);
 
+      UT_ASSERT_FALSE (GetIgnoreNextEx ());
+      UT_ASSERT_NOT_EFI_ERROR (mNonstopModeProtocol->ResetPageAttributes ());
+    }
+  } else if (MemoryProtectionContext.TestingMethod == MemoryProtectionTestReset) {
+    if (MemoryProtectionContext.TestProgress < NUM_POOL_SIZES) {
       //
-      // At this point, the test has failed. Reset test progress so failure gets recorded.
+      // Context.TestProgress indicates progress within this specific test.
+      // The test progressively allocates larger areas to test the guard on.
+      // These areas are defined in Pool.c as the 13 different sized chunks that are available
+      // for pool allocation.
       //
-      MemoryProtectionContext.TestProgress = 0;
+      // We need to indicate we are working on the next part of the test and save our progress.
+      //
+      AllocationSize = mPoolSizeTable[MemoryProtectionContext.TestProgress];
+      MemoryProtectionContext.TestProgress++;
       SaveFrameworkState (&MemoryProtectionContext, sizeof (MEMORY_PROTECTION_TEST_CONTEXT));
-      UT_LOG_ERROR ("Pool guard failed: %p", ptr);
+      SetBootNextDevice ();
+
+      //
+      // Memory type rHardwareNxProtections to the bitmask for the PcdHeapGuardPoolType,
+      // we need to RShift 1 to get it to reflect the correct EFI_MEMORY_TYPE.
+      //
+      Status = gBS->AllocatePool ((EFI_MEMORY_TYPE)MemoryProtectionContext.TargetMemoryType, AllocationSize, (VOID **)&ptr);
+
+      if (EFI_ERROR (Status)) {
+        UT_LOG_WARNING ("Memory allocation failed for type %a of size %x - %r\n", MEMORY_TYPES[MemoryProtectionContext.TargetMemoryType], AllocationSize, Status);

```

Comment:
It seems to be pre-existing throughout the file but it's nice to prefix hex values with `0x` when printing.

Reply:
I can turn this into a rebase PR and change that print along with the ARRAY_SIZE change above if you'd like

Reply:
If it's not too much trouble. I think that would be nice to have since we want to particularly expand usage of this test.

Reply:
Added 0x

Reply:
This print was removed in the restructure